pkg/proxy/handlers/spacelister_list_test.go

```diff
@@ -22,127 +23,218 @@ import (
 	"github.com/codeready-toolchain/registration-service/pkg/proxy/metrics"
 	"github.com/codeready-toolchain/registration-service/pkg/signup"
 	"github.com/codeready-toolchain/registration-service/test/fake"
+	"github.com/codeready-toolchain/toolchain-common/pkg/proxy"
+	"github.com/codeready-toolchain/toolchain-common/pkg/test"
+	"github.com/codeready-toolchain/toolchain-common/pkg/test/space"
 )
 
-func TestSpaceListerList(t *testing.T) {
-	fakeSignupService, fakeClient := buildSpaceListerFakes(t)
-
-	t.Run("HandleSpaceListRequest", func(t *testing.T) {
-		// given
-		tests := map[string]struct {
-			username             string
-			expectedWs           []toolchainv1alpha1.Workspace
-			expectedErr          string
-			expectedErrCode      int
-			expectedWorkspace    string
-			overrideSignupFunc   func(ctx *gin.Context, userID, username string, checkUserSignupComplete bool) (*signup.Signup, error)
-			overrideInformerFunc func() service.InformerService
-		}{
-			"dancelover lists spaces": {
-				username: "dance.lover",
-				expectedWs: []toolchainv1alpha1.Workspace{
-					workspaceFor(t, fakeClient, "dancelover", "admin", true),
-					workspaceFor(t, fakeClient, "movielover", "other", false),
-				},
-				expectedErr: "",
-			},
-			"movielover lists spaces": {
-				username: "movie.lover",
-				expectedWs: []toolchainv1alpha1.Workspace{
-					workspaceFor(t, fakeClient, "movielover", "admin", true),
-				},
-				expectedErr: "",
+func TestListUserWorkspaces(t *testing.T) {
+	tests := map[string]struct {
+		username            string
+		additionalSignups   []fake.SignupDef
+		additionalObjects   []runtime.Object
+		expectedWorkspaces  func(*test.FakeClient) []toolchainv1alpha1.Workspace
+		publicViewerEnabled bool
+	}{
+		"dancelover lists spaces with public-viewer enabled": {
+			username: "dance.lover",
+			additionalSignups: []fake.SignupDef{
+				newSignup("communitylover", "community.lover", true),
 			},
-			"signup has no compliant username set": {
-				username:        "racing.lover",
-				expectedWs:      []toolchainv1alpha1.Workspace{},
-				expectedErr:     "",
-				expectedErrCode: 200,
+			additionalObjects: []runtime.Object{
+				fake.NewSpace("communityspace", "member-1", "communitylover", space.WithTierName("appstudio")),
```

Comment:
very minor - I would keep the space name as the signup name, this way you can avoid setting the Owner at line 50 to something different, unless there is a reason I'm missing for naming it differently 
```suggestion
				fake.NewSpace("communitylover", "member-1", "communitylover", space.WithTierName("appstudio")),
```

Reply:
no big reason for that. Fixed in c913d90d3fde77676e706662c65f521576294805

I also renamed the SpaceBinding to be more expressive